sdk_version_resolver.go

```diff
@@ -0,0 +1,156 @@
+package dotnetcoresdk
+
+import (
+	"fmt"
+	"os"
+	"sort"
+	"strings"
+
+	"github.com/BurntSushi/toml"
+	"github.com/Masterminds/semver"
+	"github.com/paketo-buildpacks/packit"
+	"github.com/paketo-buildpacks/packit/postal"
+)
+
+type RuntimeToSdks struct {
+	RuntimeVersion string   `toml:"runtime-version"`
+	SDKs           []string `toml:"sdks"`
+}
+
+type SDKVersionResolver struct {
+	logger LogEmitter
+}
+
+func NewSDKVersionResolver(logger LogEmitter) SDKVersionResolver {
+	return SDKVersionResolver{logger: logger}
+}
+
+func (r SDKVersionResolver) Resolve(path string,
+	entry packit.BuildpackPlanEntry,
+	stack string) (postal.Dependency, error) {
+
+	var sdkVersion string
+	if sdkVersionStruct, ok := entry.Metadata["version"]; ok {
+		sdkVersion = sdkVersionStruct.(string)
+	}
+
+	if sdkVersion == "" || sdkVersion == "default" {
+		sdkVersion = "*"
+	}
+
+	sdkConstraint, err := semver.NewConstraint(sdkVersion)
```

Comment:
Why is only one constraint added, rather than gathering all the roll forward options like in dotnet-core-runtime?